pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -136,6 +139,52 @@ func (m *managerImpl) IsUnderDiskPressure() bool {
 	return hasNodeCondition(m.nodeConditions, api.NodeDiskPressure)
 }
 
+func (m *managerImpl) createThresholdNotification(diskInfoProvider DiskInfoProvider, podFunc ActivePodsFunc, observations signalObservations) error {
+	// default to no-op notifier
+	m.thresholdNotifier = NewNoOpThresholdNotifier()
+
+	thresholds := m.config.Thresholds
+	if len(thresholds) == 0 {
+		return nil
+	}
+
+	for _, threshold := range thresholds {
+		// only enable memcg threshold notification if a hard memory eviction limit is set
+		if threshold.Signal != SignalMemoryAvailable || threshold.GracePeriod != time.Duration(0) {
```

Comment:
why special case grace period?

Are we not able to define two thresholds - one for the soft and one for that hard eviction config?  By getting notified on a soft transition, we can get more accurate grace period calculation.